public/app/features/query/components/QueryEditorRow.tsx

```diff
@@ -238,20 +238,10 @@ export class QueryEditorRow<TQuery extends DataQuery> extends PureComponent<Prop
     }
   }
 
-  waitingForDatasourceToLoad = (): boolean => {
-    // if we not yet have loaded the datasource in state the
-    // ds in props and the ds in state will have different values.
-    return this.props.dataSource.uid !== this.state.datasource?.uid;
-  };
```

Comment:
The log updateQueries cleans the query models so this should not be needed 
To me it looks like a bug in Alerting when handling change data source in that it is not clearing the query model correctly.

Related to
https://github.com/grafana/support-escalations/issues/4577

Reply:
added it back but with a better check to make ExplorePage tests work